check_for_bad_version.py

```diff
@@ -22,79 +21,65 @@
-            else:
-                looking = False
-        if looking:
-            if line.startswith("+Version:"):
-                segs1 = line.rstrip().split()
-                version = segs1[len(segs1)-1]
-                segs2 = version.split('.')
-                if len(segs2) != 3:
-                    sprint("Malformed version %s in file %s." % (version, filename))
-                    sprint("See http://bioconductor.org/developers/how-to/version-numbering/")
-                    sys.exit(1)
-                for seg in segs2:
-                    fail = False
-                    if '-' in seg:
-                        fail = True
-                    try:
-                        int(seg)
-                    except: 
-                        fail = True
-                    if fail:
-                        sprint("Malformed version %s in file %s." % (version, filename))
-                        sprint("See http://bioconductor.org/developers/how-to/version-numbering/")
-                        sys.exit(1)
-                ys = segs2[1]
-                y = int(ys)
-                mod = y % 2
-                if release and mod == 1:
-                    sprint("The y in the x.y.z version number should be even in release")
-                    sprint("in file %s." % filename)
-                    sprint("See http://bioconductor.org/developers/how-to/version-numbering/")
-                    sys.exit(1)
-                if (not release) and mod == 0:
-                    sprint("The y in the x.y.z version number should be odd in devel")
-                    sprint("in file %s." % filename)
-                    sprint("See http://bioconductor.org/developers/how-to/version-numbering/")
-                    sys.exit(1)
+if (len(diff) == 0):
+    sys.exit(0)
+
+filename = None
+looking = False
+release = None
+
+for line in diff.split("\n"):
+    if line.startswith("+++ "):
+        looking = False
+    m = re.search("\+\+\+ ((\S+)/madman/Rpacks/[^/]+/DESCRIPTION)", line)
+    if m:
+        filename = m.group(1)
+        if m.group(2) == "trunk":
+            release = False
+        elif m.group(2).startswith("branches/RELEASE_"):
+            release = True
+        else:
+            continue
+        looking = True
+        continue
+    if looking:
+        mv = re.search("\+Version:\s*(\d+)\.(\d+)\.(\d+)(.*?)\s*$", line)
+        if mv:
+            try:
+                if len(mv.group(4)):
+                    raise
+                x = int(mv.group(1))
+                y = int(mv.group(2))
+                z = int(mv.group(3))
+            except:
+                sprint("Malformed version line '%s' in file %s." % (line, filename))
+                sprint("See http://bioconductor.org/developers/how-to/version-numbering/")
+                sys.exit(1)
+
+            mod = y % 2
+            if release and mod == 1:
+                sprint("The y in the x.y.z version number should be even in release")
```

Comment:
Instead of _"The x.y.z version number should be even in release"_ , I wonder if we should be more specific and say something like _"The x.y.z version number should be even when releasing in Bioconductor"_ .  Again, just a thought.